doc/development_guide/api/epylint.rst

```diff
@@ -0,0 +1,22 @@
+=======
+epylint
```

Comment:
This whole section of `API` should probably go into `User guide`? It's more usage related than development related right?

Reply:
For me everything that is not doable by using the pylint executable with options is development. (Beside creating a file to call epylint's API is development ?).

Reply:
But `Development` is about developing `pylint` right? Since `pylint` is a developer's tool everything related to it will be "development", but I feel like this is more part of "using pylint" than "developing or contributing to pylint". We never use these APIs in our own development and you don't really need them when writing a checker.

Reply:
Right, it's still unclear what is what. How about this:
- User guides : using pylint's result from the command line
- Developer guides: using pylint's API programmatically, understanding pylint plugin system, extension, checkers, etc.
- Contributor guides : developing for pylint itself launching tests, process, etc. (also maintaining pylint : we have a release.md document that is not included in the doc yet).

Reply:
I like that! Although I would perhaps put Developers and Contributing under a more general header of "Development" like black does. That would also allow showing some of the sub-headers of Usage in the sidebar.

Reply:
I think we're almost ready to do the big table of content revamp then 😄 .

Reply:
I'm planning on adding readme and release.md first.

Reply:
I'll let you choose, but I'd prefer a PR that sets up the TOC first. It's quite hard to review content + redirects + new pages at the same time.
If we create one PR with all pages/sections we want to have and make sure redirects work we can then do the content afterwards. I would even be okay with adding some blank pages for now considering that you're working on this quite hard at the moment.